src/python/pants/backend/python/lint/yapf/rules.py

```diff
@@ -0,0 +1,155 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from dataclasses import dataclass
+from typing import Tuple
+
+from pants.backend.python.lint.python_fmt import PythonFmtRequest
+from pants.backend.python.lint.yapf.skip_field import SkipYapfField
+from pants.backend.python.lint.yapf.subsystem import Yapf
+from pants.backend.python.target_types import PythonSources
+from pants.backend.python.util_rules import pex
+from pants.backend.python.util_rules.interpreter_constraints import InterpreterConstraints
+from pants.backend.python.util_rules.pex import PexRequest, PexRequirements, VenvPex, VenvPexProcess
+from pants.core.goals.fmt import FmtResult
+from pants.core.goals.lint import LintRequest, LintResult, LintResults
+from pants.core.util_rules.config_files import ConfigFiles, ConfigFilesRequest
+from pants.core.util_rules.source_files import SourceFiles, SourceFilesRequest
+from pants.engine.fs import Digest, MergeDigests
+from pants.engine.process import FallibleProcessResult, Process, ProcessResult
+from pants.engine.rules import Get, MultiGet, collect_rules, rule
+from pants.engine.target import FieldSet, Target
+from pants.engine.unions import UnionRule
+from pants.util.logging import LogLevel
+from pants.util.strutil import pluralize
+
+
+@dataclass(frozen=True)
+class YapfFieldSet(FieldSet):
+    required_fields = (PythonSources,)
+
+    sources: PythonSources
+
+    @classmethod
+    def opt_out(cls, tgt: Target) -> bool:
+        return tgt.get(SkipYapfField).value
+
+
+class YapfRequest(PythonFmtRequest, LintRequest):
+    field_set_type = YapfFieldSet
+
+
+@dataclass(frozen=True)
+class SetupRequest:
+    request: YapfRequest
+    check_only: bool
+    inplace: bool
```

Comment:
Fwict, you can get rid of `inplace`. If `check_only=False`, that implies `inplace=True`.